src/game.py

```diff
@@ -88,17 +88,48 @@ def run(self):
 
             #handle difficulty changes during the run
             npc_accuracy = self.game_stats.get_basic_npc_accuracy()
-            if npc_accuracy < 0.1:
-                self.basic_npc_difficulty_level = min(self.difficulty_manager.basic_npc_difficulty_level + 1, self.difficulty_manager.basic_npc_max_difficulty)
+            if npc_accuracy < 0.2:
+                self.npc_difficulty_level = min(self.difficulty_manager.npc_difficulty_level + 1, self.difficulty_manager.npc_max_difficulty)
                 for npc in self.npcs:
-                    self.difficulty_manager.visit_basic_npc(npc)
+                    self.difficulty_manager.visit_npc(npc)
             player_hit_treshold = self.game_stats.player_hit_treshold()
+            bigger_hit_treshold = self.game_stats.bigger_treshold()
             if player_hit_treshold:
-                self.player_difficulty_level = min(self.difficulty_manager.player_difficulty_level + 1, self.difficulty_manager.player_max_difficulty)
-                self.basic_npc_difficulty_level = min(self.difficulty_manager.basic_npc_difficulty_level - 1, 1 )
                 self.difficulty_manager.visit_player(self.player)
                 for npc in self.npcs:
-                    self.difficulty_manager.visit_basic_npc(npc)
+                    self.difficulty_manager.visit_decrease_npc_diff(npc)
+            if bigger_hit_treshold:
+                for npc in self.npcs:
+                    self.difficulty_manager.visit_bigger_decrease_npc_diff(npc)
+            #incase npcs are destroyed, add some more
+            if not self.npcs:
+                self.npcs.append(HunterNPC(self.next_id(),
+                                  3 * self.w / 4,
+                                  3 * self.h / 4,
+                                  50,
+                                  self.vision_sensor,
+                                  self.border_sensor,
+                                  self.collision_sensor,
+                                  self.game_stats))
+
+                self.npcs.append(BasicNPC(self.next_id(),
+                                  self.w / 3,
+                                  self.h / 3,
+                                  50,
+                                  self.player_position_sensor,
+                                  self.border_sensor,
+                                  self.collision_sensor,
+                                  self.game_stats))
+                
+                self.npcs.append(BasicNPC(self.next_id(),
+                                  self.w / 5,
+                                  self.h / 5,
+                                  50,
+                                  self.player_position_sensor,
+                                  self.border_sensor,
+                                  self.collision_sensor,
+                                  self.game_stats))
+            
```

Comment:
We are not using Basic NPCs anymore, because we do not have the smart features implemented for them. Can you only spawn Hunters please?
Also, you should check if they want to spawn inside something, in that case do not spawn them. It can be done with collision sensors, and I also have a spawner class implemented, probably you could add a spawn_npc method there.

Reply:
Only hunters spawning and created a npc spawning method to item_spawner